src/WebJobs.Script/Config/WorkerConcurrencyOptions.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    public class WorkerConcurrencyOptions : Hosting.IOptionsFormatter
+    {
+        public WorkerConcurrencyOptions()
+        {
+            // Setting deafault values
+            LatencyThreshold = TimeSpan.FromSeconds(1);
+            AdjustmentPeriod = TimeSpan.FromSeconds(10);
+            CheckInterval = TimeSpan.FromSeconds(1);
+            HistorySize = 10;
+            HistoryThreshold = 1F;
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether worker concurrency
+        /// is enabled.
+        /// </summary>
+        public bool Enabled { get; set; }
+
+        /// <summary>
+        /// Gets or sets the latency threshold dictating when worker channel is overloaded.
+        /// </summary>
+        public TimeSpan LatencyThreshold { get; set; }
+
+        /// <summary>
+        /// Gets or sets the time when worker
+        /// channels will start to monitor again after adding a new worker.
+        /// </summary>
+        public TimeSpan AdjustmentPeriod { get; set; }
```

Comment:
I think you should keep AdjustmentPeriod, CheckInterval, HistorySize internal. These aren't things users should have to think about.

Reply:
Those settings will be internal, we want to expose them to give us more space for experiments after the change will be deployed to PROD.

Reply:
I don't understand why you say they'll be internal if you're exposing them publicly now. We can't remove them later.